CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatMain.lua

```diff
@@ -0,0 +1,735 @@
+			if (currentChannel) then
+				EventFolder.SayMessageRequest:FireServer(message, currentChannel.Name)
+
+				if (currentChannel.Name == "All") then
+					--// Sends signal to eventually call Player:Chat() to handle C++ side legacy stuff.
+					moduleApiTable.MessagePosted:fire(message) 
+				end
+			else
+				EventFolder.SayMessageRequest:FireServer(message, nil)
+			end
+		end
+		
+	end
+end)
+
+EventFolder.OnNewMessage.OnClientEvent:connect(function(fromSpeaker, channel, message)
+	local channelObj = ChatWindow:GetChannel(channel)
+	if (channelObj) then
+		local messageObject = MessageLabelCreator:CreateMessageLabel(fromSpeaker, message)
+		channelObj:AddMessageLabelToLog(messageObject)
+		
+		ChannelsBar:UpdateMessagePostedInChannel(channel)
+		
+		moduleApiTable.MessageCount = moduleApiTable.MessageCount + 1
+		moduleApiTable.MessagesChanged:fire(moduleApiTable.MessageCount)
+
+		DoFadeInFromNewInformation()
+
+	else
+		warn("Just received chat message for channel I'm not in [" .. channel .. "]")
+		
+	end
+end)
+
+EventFolder.OnNewSystemMessage.OnClientEvent:connect(function(message, channel)
+	channel = channel or "System"
+	
+	local channelObj = ChatWindow:GetChannel(channel)
+	if (channelObj) then
+		local messageObject = MessageLabelCreator:CreateSystemMessageLabel(message)
+		channelObj:AddMessageLabelToLog(messageObject)
+		
+		ChannelsBar:UpdateMessagePostedInChannel(channel)
+		
+		moduleApiTable.MessageCount = moduleApiTable.MessageCount + 1
+		moduleApiTable.MessagesChanged:fire(moduleApiTable.MessageCount)
+
+		DoFadeInFromNewInformation()
+
+	else
+		warn("Just received system message for channel I'm not in [" .. channel .. "]")
+		
+	end
+end)
+
+EventFolder.OnChannelJoined.OnClientEvent:connect(function(channel, welcomeMessage)
+	local channelObj = ChatWindow:AddChannel(channel)
+
+	if (channelObj) then
+		if (channel == "All") then
+			ChatWindow:SwitchCurrentChannel(channel)
+		end
+		
+		if (welcomeMessage ~= "") then
+			local messageObject = MessageLabelCreator:CreateWelcomeMessageLabel(welcomeMessage)
+			channelObj:AddMessageLabelToLog(messageObject)
+		end
+
+		DoFadeInFromNewInformation()
+	end	
+	
+end)
+
+EventFolder.OnChannelLeft.OnClientEvent:connect(function(channel)
+	ChatWindow:RemoveChannel(channel)
+
+	DoFadeInFromNewInformation()
+end)
+
+EventFolder.OnMuted.OnClientEvent:connect(function(channel)
+	-- handle
+end)
+
+EventFolder.OnUnmuted.OnClientEvent:connect(function(channel)
+	-- handle
+end)
+
+EventFolder.OnSpeakerExtraDataUpdated.OnClientEvent:connect(function(speakerName, data)
+	local speaker = SpeakerDatabase:GetSpeaker(speakerName)
+	if (not speaker) then
+		speaker = SpeakerDatabase:AddSpeaker(speakerName)
+	end
+	
+	for k, v in pairs(data) do
+		speaker[k] = v
+	end
+end)
+
+
+EventFolder.OnMainChannelSet.OnClientEvent:connect(function(channel)
+	if (ChatWindow:GetChannel(channel)) then
+		ChatWindow:SwitchCurrentChannel(channel)
+	end
+end)
+
+
+local reparentingLock = false
```

Comment:
It was supposed to be used, but it wasn't implemented. In the fix I pushed it is now used.